### PR TITLE
Add core registers for Cortex M0 and M0+

### PR DESCRIPTION
**Pull request by jacobcpeters**
This should at least partially solve issues #27 and #28. I only refactored what was already there because that should be enough for my use case. The generated output looks correct, but I can test it whenever I get some time to go back to my original project.

### CODE REVIEW

File: src/svd/cmsis.zig [hunk 1]
```diff
@@ -1,86 +1,87 @@
...
-        .version = null,
         .description = "System Control Space",
-        .base_addr = 0xe000e000,
     });
-    try db.register_properties.peripheral.size.put(db.gpa, scs, 32);
 
-    if (db.cpu) |cpu| if (!cpu.vendor_systick_config)
-        try addSysTickRegisters(db, scs);
+    if (db.instances.devices.get(device_id)) |cpu| {
+        if (cpu.properties.get("cpu.vendor_systick_config") != null)
```
Comment (mattnite) [inline src/svd/cmsis.zig]: `vendor_systick_config` needs to be evaluated, not just the null check

File: src/svd/cmsis/cortex_m0.zig [hunk 2]
```diff
@@ -1,267 +1,199 @@
...
-    const base_addr = if (peripheral.base_addr) |periph_base_addr|
-        periph_base_addr + cluster.addr_offset
-    else
-        cluster.addr_offset;
 
-    var register_fields = std.ArrayList([]Field).init(db.gpa);
-    defer register_fields.deinit();
+        const nvic_prio_bits = try parseInt(
+            u32,
+            db.instances.devices.get(device_id).?.properties.get("cpu.nvic_prio_bits").?,
```
Comment (mattnite) [inline src/svd/cmsis/cortex_m0.zig]: Please handle missing prio bits by returning an error

**Review by mattnite — CHANGES_REQUESTED** [review]
looking great! just some small details. This is much appreciated

**Comment by jacobcpeters** [comment]
If `cpu.nvic_prio_bits` is missing, there should be an error earlier in the logic. However, I did make the check explicit using the same error as earlier in svd.zig:87

In the check for `cpu.vendor_systick_config`, as far as I can tell from my searches about XML, the only proper values for an `xs:boolean` are "true" and "false", both specifically lowercase, as well as "1" and "0". Any other value will return an error for the bad representation.

**Comment by mattnite** [comment]
CI will dial but that’s just from zig master churn, code looks great to me, thanks!

**Pull request merged**